### PR TITLE
Unify and update scripts

### PR DESCRIPTION
**Pull request by allanlw**
This creates a unified QEMU script in the scripts directory, that can be used for development and for the test suite.

### CODE REVIEW

File: script/QEMU.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+while true; do
+  case "$1" in
+    -h|--help)
+      echo "usage: $0 [opts] programs..."
+      echo ""
+      echo "  -h, --help          Print this help message"
+      echo "  -d, --drive=FILE    Drive to use (default=temp_drive)"
+      echo "  -S, --wait          Pause CPU on startup"
+      echo "  -s, --serial=FILE   Output serial to FILE instead of stdout"
+      echo "  -c, --coverage=FILE Output unique hit IP values to FILE"
```
Comment (gsilvis) [inline script/QEMU.sh]: maybe 'instructions' or similar---this looks like 'IP addresses' to me

File: script/QEMU.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      shift; shift;;
+    --nographic)
+      NOGRAPHIC=1
+      shift;;
+    -d|--drive)
+      DRIVE="$2"
+      shift; shift;;
+    -c|--coverage)
+      COVERAGE="$2"
+      shift; shift;;
```
Comment (gsilvis) [inline script/QEMU.sh]: In Perl, 'shift' has a value---looking at this code makes me miss that :(

File: script/QEMU.sh [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    --) shift; break;;
+    *) break ;;
+  esac
+done
+
+MODULES="$(join_by , george.multiboot "$@")"
+
+QEMU_ARGS=" -kernel bootloader.multiboot -initrd $MODULES"
```
Comment (gsilvis) [inline script/QEMU.sh]: All of the arguments are one-per-line, so I'd split this one up too

File: README.md [hunk 4]
```diff
@@ -5,7 +5,9 @@ sufficient.
 
 To run, execute:
 
-    ./script/TEST-QEMU.sh & ./script/TEST-GDB.sh
+    ./script/QEMU.sh userland/calc.bin
+
+Or whatever other userland program(s) you would like.
```
Comment (gsilvis) [inline README.md]: Can you add an example that uses -s?

File: script/QEMU.sh (hunk 1 above)
Comment (allanlw) [inline script/QEMU.sh]: Replaced with `kernel %rip`

File: script/QEMU.sh (hunk 2 above)
Comment (allanlw) [inline script/QEMU.sh]: Actually bash does apparently have this.

File: script/QEMU.sh (hunk 3 above)
Comment (allanlw) [inline script/QEMU.sh]: Done.

File: README.md (hunk 4 above)
Comment (allanlw) [inline README.md]: done.

**Comment by allanlw** [comment]
All of the things that George complained about have been fixed.

**Pull request merged**